Apps/APN-UIKit/APN UIKit/View/DashboardViewController.swift

```diff
@@ -1,88 +1,106 @@
+        versionsLabel.text = getMetaData()
+    }
     func addUserInteractionToImageViews() {
         settings.addTapGesture(onTarget: self, #selector(DashboardViewController.settingsTapped))
-        userDetail.addTapGesture(onTarget: self, #selector(DashboardViewController.userDetailTapped))
     }
-
     deinit {
         NotificationCenter.default.removeObserver(self)
     }
-
     func addNotifierObserver() {
         NotificationCenter.default.addObserver(
             self,
-            selector: #selector(routeToDeepLinkScreen(notification:)),
-            name: Notification.Name("showDeepLinkScreenOnDashboard"),
+            selector: #selector(deepLinkRouteToSettings(notification:)),
+            name: Notification.Name("showSettingsScreenOnDashboard"),
             object: nil
         )
     }
-
     @objc
-    func routeToDeepLinkScreen(notification: Notification) {
+    func deepLinkRouteToSettings(notification: Notification) {
         if let userInfo = notification.userInfo as? [String: String] {
-            dashboardRouter?.routeToDeepLinkScreen(withInfo: userInfo)
+            dashboardRouter?.routeToSettings(userInfo)
         }
     }
 
     @objc func settingsTapped() {
-        dashboardRouter?.routeToSettings()
+        dashboardRouter?.routeToSettings(nil)
     }
 
-    @objc func userDetailTapped() {
-        let userDetail = "Name - " + storage.userName! + "\n\nEmailId - " + storage.userEmailId!
-        showAlert(withMessage: userDetail)
+    func addAccessibilityIdentifiersForAppium() {
+        setAppiumAccessibilityIdTo(settings, value: "Settings")
+        setAppiumAccessibilityIdTo(randomEventButton, value: "Random Event Button")
+        setAppiumAccessibilityIdTo(customEventButton, value: "Custom Event Button")
+        setAppiumAccessibilityIdTo(sendDeviceAttributesButton, value: "Device Attribute Button")
+        setAppiumAccessibilityIdTo(sendProfileAttributesButton, value: "Profile Attribute Button")
+        setAppiumAccessibilityIdTo(showPushPromptButton, value: "Show Push Prompt Button")
+        setAppiumAccessibilityIdTo(logoutButton, value: "Log Out Button")
+    }
+    func setUserDetail() {
+        if let email = storage.userEmailId {
+            userInfoLabel.text = email
+        }
     }
 
     // MARK: - Actions
 
     @IBAction func logoutUser(_ sender: UIButton) {
         storage.userEmailId = nil
-        storage.userName = nil
         CustomerIO.shared.clearIdentify()
         dashboardRouter?.routeToLogin()
     }
 
     @IBAction func sendRandomEvent(_ sender: UIButton) {
-        let randomEventName = String.generateRandomString(ofLength: 10)
-        CustomerIO.shared.track(name: randomEventName)
-        showAlert(withMessage: "Random event '\(randomEventName)' tracked successfully")
+        let randomInt = Int.random(in: 0..<3)
+        let randomEventInfo = randomData[randomInt]
+        guard let name = randomEventInfo["name"] as? String else {
+            return
```

Comment:
do we show any error if this is not present?

Reply:
Name will always be present for random data. This check is just to unwrap the name data inside `randomData`.